Dismemberment

phil235 opened the pull request:
port from Hippie code, based on RemieRichard's work (https://github.com/HippieStationCode/HippieStation13/pull/2190 and https://github.com/tgstation/-tg-station/pull/6335). 
Big thanks to @RemieRichards and @Crystalwarrior

What it does:
- your legs, arms and head can now be separated from your body
- Things that can dismember you: explosions, big sharp weapons
- Fixes COLDRES being a define that is defined TWICE and with different value.
- Renamed the list of bodyparts to "bodyparts" instead of the stupid "organs"
- Repathed limb and internal organ items.
- Limbs that are attached to you to replace a missing limb are not sync'd with your original limbs (e.g. getting hulk will not make this new limb green). But regenerated limbs (admin heal, changeling fleshmend) are sync'd.
- Changeling can survive a decapitation, grow all bodyparts back with fleshmend, or just one arm with arm_blade/shield thing.
- Limbs lost while a hulk are not green.
- You cannot handcuff/legcuff people with less than 2 arms/legs.
- Dismember a chest does not drop it, but spills its organs on the ground.
- If you have cyber implants inside a limb, you lose those when the limb is dismembered. The implants are stored inside the dropped limb.
- You can use a sharp weapon to retrieve the organs and items inside a dropped bodypart.
- limb-attachedchainsaw is safely detached when losing the arm.
- you cannot pick things up with your right hand if missing your right arm.
- having one leg makes you move slower.
- <b>having no legs makes you crawl lying down, you cannot pick up object unless buckled to a chair.</b>
- You can amputate bodyparts through surgery.
- You can replace missing bodyparts with robotic ones through surgery.
- You can insert a brain inside a robotic head put on a headless body.
- The bodyparts list is now a carbon level var (although only humans can be dismembered yet).
- Missing a bodypart is indicated on your healthdoll.

The things that were not done in this PR and remains to be added:
- decapitated lizard/purrbation human head showing horns, frills, ears, etc.
- carbon dismemberment
- inhands sprites for bodyparts
- incorporating lizard tail severing to dismemberment.
- Making all "for(var in internal_organs)" into istypeless loops.(since there are now only organs inside that list.

<b>The hidden nerfs of the PR:
- synth and military synth species are broken (but they already were, so whatever)
- plasmamen's skinbone icon alternative is removed.
- cold resistance mutation sprite is now just a glowing ball behind your mob centered around the chest.
- adminspawned limbs are invisible.
- head augmentation icon is changed, it now looks like a full robotic head, instead of just some metal around your ear and at the back of the head.
- Lizard body markings sprites now only cover the chest.</b>

:cl:
rscadd: DISMEMBERMENT! Humans can now lose their bodyparts. Explosions and being attacked with a heavy sharp item on a specific bodypart can cause dismemberment.
rscadd: You can't be handcuffed or legcuffed if you're missing an arm or leg. You can't use items when you have no legs, your arms are too busy helping you crawl, unless your are buckled to a chair. You're slower when missing a leg.
rscadd: You lose organs and items if the bodypart they are inside gets cut off. You can always retrieve them by cutting the dropped bodypart with a sharp object.
rscadd: Medbay can replace your missing limbs with robotic parts via surgery, or amputate you.
rscadd: Changelings do not die when decapitated! They can regrow all their limbs with Fleshmend, or just one arm with Arm Blade or Organic Shield.
rscadd: Your chest cannot be dropped, but it will spill its organs onto the ground when dismembered.
/:cl:

phil235 committed:
Dismemberment port from Hippie code, based on RemieRichard's work. Big thanks to RemieRichards and crystalwarrior.

Razharas commented:
Nice

GunHog commented:
# ULTRA HEAVY BREATHING

RemieRichards commented on code/game/gamemodes/changeling/powers/mutations.dm:
```diff
@@ -38,10 +38,18 @@
+	else
+		limb_regen = user.regenerate_limb("r_arm", 1)
+	if(limb_regen)
+		user.visible_message("<span class='warning'>[user]'s missing arm reform, making a loud, grotesque sound!</span>", "<span class='userdanger'>Your arm regrow, making a loud, crunchy sound and giving you great pain!</span>", "<span class='italics'>You hear organic matter ripping and tearing!</span>")
```
reforms

ChangelingRain replied:
regrows

coiax replied:
regenerates

RemieRichards replied:
@coiax we're not suggesting alternate words, we're pointing out the ones that are missing ```s```'s

coiax replied:
s/ and giving you great pain!/. It hurts a considerable deal./

RemieRichards commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -29,8 +29,13 @@
 		var/mob/living/carbon/human/H = M
+		affecting = H.get_bodypart(check_zone(user.zone_selected))
+		if(!affecting) //Missing limb?
+			user << "<span class='warning'>[H] doesn't have [parse_zone(user.zone_selected)]!</span>"
```
\a

RemieRichards commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -71,7 +77,10 @@
-		var/obj/item/organ/limb/affecting = H.get_organ(check_zone(user.zone_selected))
+		affecting = H.get_bodypart(check_zone(user.zone_selected))
+		if(!affecting) //Missing limb?
+			user << "<span class='warning'>[H] doesn't have [parse_zone(user.zone_selected)]!</span>"
```
\a

Razharas commented on code/_onclick/hud/screen_objects.dm:
```diff
@@ -89,6 +90,8 @@
 		var/state = (slot_id == slot_r_hand) ? "markus" : "gabrielle"
 		handcuff_overlay = image("icon"='icons/mob/screen_gen.dmi', "icon_state"=state)
+	if(!blocked_overlay)
+		blocked_overlay = image("icon"='icons/mob/screen_gen.dmi', "icon_state"="blocked")
```
Already discussed this n coderbus but please make it remove the hand icon completely for absolute immersion instead of blocked overlay

phil235 replied:
Meh, will do in another PR.

RemieRichards commented on code/modules/admin/verbs/debug.dm:
```diff
@@ -362,7 +362,7 @@ var/list/TYPES_SHORTCUTS = list(
 	/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/launcher/missile_rack = "MECHA_MISSILE_RACK",
 	/obj/item/mecha_parts/mecha_equipment = "MECHA_EQUIP",
-	/obj/item/organ/internal = "ORGAN_INT",
+	/obj/item/organ = "ORGAN_INT",
```
probably just ORGAN now.

TrustyGun commented:
im ready

RemieRichards commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -150,15 +147,18 @@
 	take_overall_damage(b_loss,f_loss)
-
+	//attempt to dismember bodyparts
+	for(var/obj/item/bodypart/BP in bodyparts)
```
this should be an istypeless forloop.

coiax replied:
when is a typed for loop acceptable?

RemieRichards replied:
with mixed data.

coiax replied:
wait, it's acceptable with mixed data? I thought it was the other way around?

RemieRichards replied:
No, a typed for loop runs automatic ```istype()```s on the contents of the list, so filters what you want out of mixed data.
for homogeneous data there's not point ```istype()```ing every element.

RemieRichards commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -185,6 +191,11 @@
 	H.remove_overlay(HAIR_LAYER)
+
+	var/obj/item/bodypart/head/HD = H.get_bodypart("head")
+	if(!HD || HD.status == ORGAN_ROBOTIC) //Decapitated or robotic head
```
I think aug'd people should still get hair tbh.

phil235 replied:
The issue is that replacing a decapitated head with a robotic one use the same code as head augmenting, so you can't have hair with augments but not with robotic prosthetic head.

coiax replied:
Wig slot when

Supermichael777 replied:
Okay but i don't think prosthetic heads will catch on.

RemieRichards commented on code/modules/surgery/bodyparts/dismemberment.dm:
```diff
@@ -0,0 +1,314 @@
+	var/direction = pick(cardinal)
+	var/t_range = rand(2,max(throw_range/2, 2))
+	var/turf/target_turf = get_turf(src)
+	for(var/i = 1; i < t_range; i++)
```
```for(var/i in 1 to t_range-1)```

RemieRichards commented on code/modules/surgery/bodyparts/dismemberment.dm:
```diff
@@ -0,0 +1,314 @@
+
+/obj/item/bodypart/chest/dismember()
+	var/mob/living/carbon/human/H = owner
+	if(!istype(H) || !H.dna.species:has_dismemberment) //human's species don't allow dismemberment
```
```:```!

GunHog replied:
Colon alert!! Launch the missles!!

phil235 replied:
fuck, that's a shitty copypasta from me :confounded:

bgobandit commented:
yo can you fold lizard tail severing into this

Razharas commented on code/game/gamemodes/changeling/powers/fleshmend.dm:
```diff
@@ -30,6 +30,11 @@
 			H.restore_blood()
 			H.remove_all_embedded_objects()
+			if(H.get_missing_limbs())
+				playsound(get_turf(H), 'sound/effects/blobattack.ogg', 30, 1)
```
That would sound more silly than cool

Razharas commented on code/game/objects/items.dm:
```diff
@@ -572,3 +574,6 @@ obj/item/proc/item_action_slot_check(slot, mob/user)
 	return sharpness
+
+/obj/item/proc/can_dismember()
+	return sharpness && w_class >= 3
```
Maybe make it return actual probability so you can tweak it per item if needed instead of just binary state of possibility?

coiax replied:
I'm going to assume that this is binary on/off flag indicating the possibility for dismemberment, like sharpness.

Razharas replied:
I know, and i suggest to make it more than that

coiax replied:
It should be a separate function/var though. In the same way that a wirecutter is a substitute tool for clamping bleeders, but isn't always successful.

PKPenguin321 commented:
Based phil

Razharas commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -29,8 +29,13 @@
+		affecting = H.get_bodypart(check_zone(user.zone_selected))
+		if(!affecting) //Missing limb?
+			user << "<span class='warning'>[H] doesn't have [parse_zone(user.zone_selected)]!</span>"
+			return
```
If we are at it maybe adding the check for covering by clothes would be nice

phil235 replied:
that's for another PR though. Looks like I forgot a "\a" here as well.

KorPhaeron commented:
RULES OF NATURE

Dorsisdwarf commented:
Priority: Beyond Critical

Bawhoppen commented:
>lzimann confused this

BAN HE

coiax commented on code/modules/surgery/surgery.dm:
```diff
@@ -10,7 +10,7 @@
 	var/ignore_clothes = 0										//This surgery ignores clothes
 	var/obj/item/organ/organ									//Operable body part
-
+	var/requires_missing_bodypart = 0							//Surgery available only when a bodypart is present, or only when it is missing.
```
Why not FALSE instead of 0? Also, it took me some parsing to work out what this variable was for, not sure if there's a better way of phrasing it?

phil235 replied:
I just copypasted that var, I'll change it to use TRUE/FALSE. I'm taking suggestion for a better var name.

coiax replied:
To be honest, changing it to "requires_bodypart" would probably be better. Less double negatives. If there were agnostic surgeries, we could get away with adding another var "requires_absence", but that just means that you could theoretically get the impossible case of a surgery both requiring the presence and absence of a limb.

coiax commented on code/modules/surgery/helpers.dm:
```diff
@@ -22,11 +22,15 @@
+							continue
+						if(S.requires_organic_bodypart && affecting.status == ORGAN_ROBOTIC)
+							continue
+					else if(H && !S.requires_missing_bodypart) //human with no limb in surgery zone when we need a limb
```
Are there surgeries that are bodypart presence agnostic?

phil235 replied:
I don't think so. the var name isn't great so if you have better names I'm all ears.

Bawhoppen commented:
if you're going to merge lizard tails into this, merge catears and cattails in too (and give them appropriate surgery)

coiax commented on code/modules/surgery/bodyparts/head.dm:
```diff
@@ -0,0 +1,163 @@
+
+/obj/item/bodypart/head
+	name = "head"
+	desc = "what a way to get a head in life..."
```
s/what/What/

phil235 replied:
I forgot to remove that line.

coiax replied:
Actually change description to "How's your head?"

Cheridan replied:
"Didn't make sense not to live for fun, your brain gets smart but your head gets dumb."

phil235 commented:
@Bawhoppen I'm not changing lizard tail code in this PR.

coiax commented on code/modules/surgery/bodyparts/dismemberment.dm:
```diff
@@ -0,0 +1,314 @@
+		target_turf = new_turf
+		if(new_turf.density)
+			break
+	throw_at_fast(target_turf, throw_range, throw_speed)
```
So there's no slow version of dismemberment? Where the limb kind of just falls off and flops on the ground?

PKPenguin321 replied:
It's more fun this way imo

phil235 replied:
When you want to drop a limb without the screaming and throwing, you use "drop_limb()".

coiax commented on code/modules/surgery/bodyparts/dismemberment.dm:
```diff
@@ -0,0 +1,314 @@
+
+
+//Dismember a limb
+/obj/item/bodypart/proc/dismember()
```
This really needs to spray blood/friendship/*insert internal body fluid here* given that you're cutting off a limb.

phil235 replied:
I added some blood on dismember.

coiax commented on code/modules/surgery/bodyparts/bodyparts.dm:
```diff
@@ -0,0 +1,337 @@
+
+/obj/item/bodypart/chest
+	name = "chest"
+	desc = "why is it detached..."
```
Replace with "It's impolite to stare at a person's chest."

phil235 replied:
I think this was supposed to be a debug message, you're not supposed to see chest bodyparts, ever.

coiax commented on code/modules/surgery/bodyparts/bodyparts.dm:
```diff
@@ -0,0 +1,337 @@
+
+/obj/item/bodypart/l_arm
+	name = "left arm"
+	desc = "why is it detached..."
```
Replace with "Did you know that the word 'sinister' stems originally from the Latin 'sinestra' (left hand), because the left hand was supposed to be possessed by the devil? This arm appears to be possessed by no one though."

PKPenguin321 replied:
I don't like that one it's too wordy
If we need a witty description for the hands/arms just say something like "This reminds you of your girlfriend."

coiax commented on code/modules/surgery/bodyparts/bodyparts.dm:
```diff
@@ -0,0 +1,337 @@
+
+/obj/item/bodypart/r_arm
+	name = "right arm"
+	desc = "why is it detached..."
```
Replace with "Over 87% of humans are right handed. That figure drops much lower when you're missing your right arm."

TrustyGun replied:
also all of these "why is it detached..."'s could just be defined on /obj/item/bodypart

coiax commented on code/modules/surgery/bodyparts/bodyparts.dm:
```diff
@@ -0,0 +1,337 @@
+
+/obj/item/bodypart/l_leg
+	name = "left leg"
+	desc = "why is it detached..."
```
Replace with "Some athletes prefer to tie their shoelaces in a specific order for good luck. In this instance, it probably would not have helped."

coiax commented on code/modules/surgery/bodyparts/bodyparts.dm:
```diff
@@ -0,0 +1,337 @@
+
+/obj/item/bodypart/r_leg
+	name = "right leg"
+	desc = "why is it detached..."
```
Replace with "You put your right leg in, your right leg out. In, out, in, out, shake it all about. And apparently then it detaches. The hokey cokey has certainly changed a lot since space colonisation."

phil235 commented on code/modules/surgery/cavity_implant.dm:
```diff
@@ -13,32 +13,31 @@
-			target.internal_organs += tool
-			tool.loc = target
+			CH.cavity_item = tool
+			tool.loc = null
```
We no longer shove the item in the organs list and inside the mob.

Core0verload replied:
Wouldn't that cause problems with implanted bombs, grenades and such?

phil235 replied:
fuck, you're right

GunHog commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -150,15 +147,18 @@
-
+	//attempt to dismember bodyparts
+	for(var/obj/item/bodypart/BP in bodyparts)
+		if(prob(50/severity) && BP.body_zone != "head" && BP.body_zone != "chest")
```
Bomb armor should be a factor here!

phil235 replied:
Probably work for another PR.

phil235 committed:
Some adjustments, following suggestions in the PR discussion.

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Dismemberment

Conflicts:
	code/datums/martial.dm
	code/game/gamemodes/changeling/powers/fleshmend.dm
	code/modules/mob/living/carbon/carbon.dm

phil235 committed:
always gotta fail the merged conflict fixing.

PKPenguin321 commented:
>limb-attachedchainsaw is safely detached when losing the arm.

Thanks for taking care of this. Once this PR is in I will try to see if it's feasible to make the chainsaw an actual arm replacement instead of a weird abstract item.

phil235 commented:
Noooo, metastation, whyyy

PKPenguin321 commented:
Easy fix Phil, just open meta as a txt file, ctrl+f the conflicting heart path, manually change it to the new path

phil235 committed:
Fixes organ item path change on metastation map.

Iamgoofball commented:
:+1: I'll make sure goonmed is updated to work with this.

Bawhoppen commented:
fucking bird

Incomptinence commented:
Dismember birdboat the time has come.

phil235 committed:
and birdstation...

phil235 commented:
fuck me

phil235 committed:
and birdStation again...

phil235 commented:
Travis, don't fail me this time

phil235 committed:
Birdboat...

Bawhoppen commented:
Birdboat passed! :tada: :tada: :tada: :tada:

phil235 committed:
fixes the healthdoll.

Incoming5643 commented:
It doesn't have to be done here, but a species flag for being able to regrow limbs or survive without a head would be nice.

phil235 committed:
Fixes the healthdoll not being updated right away after an amputation.

Robustin commented:
Ok actual feedback:

- Shaking someone with no limbs (help intent) makes them do some wierd shit that makes them look like they're growing out of the floor

- People with no legs move too fast. It's like +1 slowdown when you have no limbs, which is faster than someone with all their limbs but missing a good chunk of health

Armhulen commented:
Forget I said this if this is already the case but you should not be able to move if you have no arms and no legs
I want to turn the whole station into chicken tendies but if they move hyper fast like soccer balls i can't do that now can i

PKPenguin321 commented:
hol up hol up u tellin me this is already playable n shit?

Lumipharon commented:
>Oh boy time for another fake meme-
>Its real

Excellent.

Incoming5643 commented:
the tongue organ is being stored in the body and not the head

bgobandit commented:
Bugs so far:

* Admin-healing someone whose head has been chopped off doesn't work -- they grow a new head but the body can't do anything
* Beheading someone with purrbation deletes the kitty ears and tail

Incoming5643 commented:
This isn't your fault and is gonna be a bitch to fix for whoever takes it up but all the lizard tail wag sprites don't have sprites to account for the lack of legs, so they disapear behind the lack of leg(s). They're actually stored on a layer behind the mob so they could have had those sprites without the legs bits cut out, but that's just not what's in the code.

Maybe if we're lucky @WJohn has versions of the intact tail wag

Incoming5643 commented:
Also I can't put my finger on it but something changes with the human sprites. At least the female ones.

PKPenguin321 commented on code/modules/surgery/prosthetic_replacement.dm:
```diff
@@ -0,0 +1,48 @@
+
+
+/datum/surgery_step/add_prosthetic
+	name = "add prothetic"
```
prosthetic

PKPenguin321 commented:
purrbation and lizard shit should be low priority, for now we should just focus on the main dismemberment. too many additions at once is what killed the last PR.

@phil235 Super edge case bug I found: Trying to engage a hardsuit helmet when you have no head properly fails to equip the helmet, but the hardsuit still thinks that the helmet is up

KorPhaeron commented:
How the fuck are you engaging a hardsuit helmet while headless in the first place?

PKPenguin321 commented:
changelings survive without a head. edge case as fuck like i said

oh also, every time you chop off a limb that isn't the chest, the message pops up like:
>Urist McName's **the** head has been violently dismembered!

there is an extra "the" between their name and the limb that was dismembered

Incoming5643 commented:
body markings are rendering underneath the torso

PKPenguin321 commented on code/game/gamemodes/changeling/powers/mutations.dm:
```diff
@@ -38,10 +38,18 @@
+	else
+		limb_regen = user.regenerate_limb("r_arm", 1)
+	if(limb_regen)
+		user.visible_message("<span class='warning'>[user]'s missing arm reforms, making a loud, grotesque sound!</span>", "<span class='userdanger'>Your arm regrow, making a loud, crunchy sound and giving you great pain!</span>", "<span class='italics'>You hear organic matter ripping and tearing!</span>")
```
>Your arm regrow, making a loud, crunchy sound and giving you great pain!

regrow*s*
should probably just get rid of "and giving you great pain" it makes the sentence too wordy and sounds bad

Iamgoofball replied:
how is army formed?
how is army formed?
how lung get rearmy?

PKPenguin321 commented:
turning into a monkey and back again regrows organic fleshly limbs. if you're missing a limb/have a prosthetic on that limb and monkey->unmonkey, you get your original limb back.

also limbs should have some force/throwforce so you can tear off a dude's arm and beat him to death with it

Lumipharon commented:
we animorphs now.

Steelpoint commented:
>cut off your arm
>never get handcuffed 

New meta strat confirmed.

Overall its nice to see dismemberment.

xxalpha commented:
>>cut off your arm
>>never get handcuffed

It's going to happen.

Fox-McCloud commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -150,15 +147,19 @@
+	for(var/X in bodyparts)
+		var/obj/item/bodypart/BP = X
+		if(prob(50/severity) && BP.body_zone != "head" && BP.body_zone != "chest")
+			BP.dismember()
```
May want to limit the amount of parts that can dismember based on explosion severity. You really shouldn't lose all your limbs form a light, 30 damage explosion just because of probability check. LIkewise, bomb armor should probably reduce or eliminate the chances of this happening.

GunHog replied:
> LIkewise, bomb armor should probably reduce or eliminate the chances of this happening.

The comment where I also requested this was hidden by a different change in code. Phil stated that he will handle this in a separate PR.

comma commented on code/modules/surgery/bodyparts/bodyparts.dm:
```diff
@@ -0,0 +1,334 @@
+	if(brute_dam > 0)
+		user << "<span class='warning'>This limb has [brute_dam > 30 ? "severe" : "minor"] bruising.</span>"
+	if(burn_dam > 0)
+		user << "<span class='warning'>This limb has [burn_dam > 30 ? "severe" : "minor"] bruising.</span>"
```
Probably meant 'burns' here

phil235 commented:
>Shaking someone with no limbs (help intent) makes them do some wierd shit that makes them look like they're growing out of the floor

Will fix


>People with no legs move too fast. It's like +1 slowdown when you have no limbs, which is faster than someone with all their limbs but missing a good chunk of health

Will nerf the speed

>the tongue organ is being stored in the body and not the head

Will fix


>Admin-healing someone whose head has been chopped off doesn't work -- they grow a new head but the body can't do anything

Intended. When a non changeling is decapitated the player becomes a brain mob inside the head. Healing the headless corpse won't bring the player back into the body.


>Beheading someone with purrbation deletes the kitty ears and tail

I was aware of that, I'm too lazy to write the code for this. I couldn't reproduce the tail disappearing though.

>Plasmaman limbs don't catch fire

That's a bug I guess, but hard to do so kind of a feature request.

>You can't behead monkeys, xenos or anything but humans

Yep, carbon dismemberment need to be coded, but not in this PR.

>This isn't your fault and is gonna be a bitch to fix for whoever takes it up but all the lizard tail wag sprites don't have sprites to account for the lack of legs, so they disapear behind the lack of leg(s). They're actually stored on a layer behind the mob so they could have had those sprites without the legs bits cut out, but that's just not what's in the code.

I can fix those tail sprites




>purrbation and lizard shit should be low priority, for now we should just focus on the main dismemberment. too many additions at once is what killed the last PR.

Indeed

>Super edge case bug I found: Trying to engage a hardsuit helmet when you have no head properly fails to equip the helmet, but the hardsuit still thinks that the helmet is up

Will fix

>oh also, every time you chop off a limb that isn't the chest, the message pops up like:
Urist McName's the head has been violently dismembered!

Will fix

>body markings are rendering underneath the torso in all directions and underneath the head while facing south

Will fix



>turning into a monkey and back again regrows organic fleshly limbs. if you're missing a limb/have a prosthetic on that limb and monkey->unmonkey, you get your original limb back.

I'm aware of that, it will be fixed when carbon dismemberment is implemented.

>also limbs should have some force/throwforce so you can tear off a dude's arm and beat him to death with it

Sure, why not.

>May want to limit the amount of parts that can dismember based on explosion severity. You really shouldn't lose all your limbs form a light, 30 damage explosion just because of probability check. LIkewise, bomb armor should probably reduce or eliminate the chances of this happening.

I'll reword the formula.

coiax commented:
Requested Features
- Heads on spikes (make sure to check ash walkers have access to spike crafting or alternative spikes)
- Wheelchairs (seeing legless people wandering around dragging a chair behind them)
- Being able to cut off the chaplain's Godhand and use it

KorPhaeron commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -951,23 +999,45 @@
+		return 0 //item force is zero
+
+	//dismemberment
+	if(affecting.get_damage() >= affecting.max_damage)
```
Do you think changing the formula to 

affecting.get_damage >= affecting.max_damage - I.armour_penetration/2 

Would be fun? It would mean that eswords slice off limbs at 50 health. I will probably make a PR for this at some point if you dont.

PKPenguin321 commented:
To anybody debugging this, super sharpening blocks are really handy for making weapons that dismember quickly. Just go "spawn sharp" and hit enter twice then use literally any object on the block and you should be good to go.

RemieRichards commented:
>synth and military synth species are broken (but they already were, so whatever)
>(but they already were, so whatever)

Not true, but w/e.

Fox-McCloud commented:
Other suggestion: when you gib, have it throw all their limbs everywhere.

Niknakflak commented:
Synths and military synths were buggy but not broken yea

RemieRichards commented:
Nobody told me anything about them was buggy, and from my testing nothing really was, anything that was wrong with them that I saw I fixed (Except for the cloning breaking it, and the Magic Mirror but only because it's Admin only, and Synths being cloned is kinda rare and my "fix" for that would really just be adding NOCLONE to the species)

phil235 commented:
Well, I couldn't get them to show their real skin through their disguise when injured enough, so I assumed it was broken.

RemieRichards commented:
@phil235 that was 100% functional the last time I saw it, I put effort into it as it's pretty much the biggest feature about them.

Anyways this is getting a bit off topic.

phil235 committed:
Fixes lizard body markings appearing behind the bodypart layer.
Fixes tail sprite when missing a leg and facing south.
Adds an admin option in human VV to remove a body part.
Fixes typos.
Fixes being able to toggle hardsuit and wintercoat helmet despite having no head.
Changes the dismemberment chance from explosion: bomb armor helps against it, and the maximum number of limbs that you can lose in one explosion is capped depending on the explosion severity.
Missing 1 or 2 legs makes you significantly slower. If without legs, missing an arm makes you slower. You can no longer move when missing all four limbs.
Bodyparts have now force 3 and throwforce 3
Tongues are now properly put in decapitated heads.
Shaking a legless person no longer gives weird results.

phil235 commented:
I added an admin option in human VV to remove a body part.

I changed the dismemberment chance from explosion: bomb armor helps against it, and the maximum number of limbs that you can lose in one explosion is capped depending on the explosion severity.

Missing 1 or 2 legs now makes you significantly slower. If without legs, missing an arm makes you even slower. You can no longer move when missing all four limbs.

Dorsisdwarf commented:
hnnnnnnnnnnggh

Iamgoofball commented:
pls merge asap so I can fix the merge conflicts

Cheridan commented:
YOLO!!!

PKPenguin321 commented:
WE DID IT REDDIT

Dorsisdwarf commented:
:hooray:

Crystalwarrior commented:
Noice.

Incoming5643 commented:
Hey @phil235 did you have individual on_mob bodyparts for the races that ended up with NODISMEMBER in this pull? I kinda want to add dismemberment to slimepeople/jellypeople with regenerating limb powers but manually making the limbs is tedious stuff.

phil235 commented:
nope
